pkg/wireguard/agent/agent_test.go

```diff
@@ -80,12 +84,15 @@ func containsIP(allowedIPs []net.IPNet, ipnet *net.IPNet) bool {
 }
 
 func (a *AgentSuite) TestAgent_PeerConfig(c *C) {
-	ctx, cancel := context.WithCancel(context.Background())
-	defer cancel()
-	ipCache := ipcache.NewIPCache(&ipcache.Configuration{
-		Context: ctx,
-	})
-	defer ipCache.Shutdown()
+	lc := &hive.DefaultLifecycle{}
+	defer lc.Stop(context.TODO())
+
+	idResolver, idPromise := promise.New[cache.IdentityAllocator]()
+	idResolver.Resolve(nil)
+	repoResolver, repoPromise := promise.New[*policy.Repository]()
+	repoResolver.Resolve(nil)
```

Comment:
Hmm, maybe we can add a constructor for resolved promise, or do vararg trick so it can be given optionally (`New[T](v ...T)`).

Reply:
I feel that this is quite niche, it isn't often we have to make pre-resolved promises